.github/workflows/update-manifest.yml

```diff
@@ -22,40 +22,110 @@ on:
         type: string
         required: true
         default: "staging"
+  workflow_run:
+    workflows: ['Create and Commit Next-Build Docker Image']
+    types: [completed]
+    branches: [main]
 
 jobs:
   update-manifest:
+    if: ${{ !(github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'failure') }}
```

Comment:
Same as before: if this is triggered by workflow_run and that was a failure, do not run.